Address additional feedback to CapTP spec (#60)
ocapn/ocapn: draft-specifications/CapTP Specification.md

# Promises

Promises are a key part of CapTP. They are used to represent a value which is
not yet known. Promises without a value are said to be unresolved, they can
become resolved by being `fulfill`ed with a value (including another promise),
or broken (`break`) with an error.

Promises are often created by sending an `op:deliver` message, where they
represent the eventual value of the response. They can be chained together in
what is called [Promise Pipelining](#promise-pipelining), whereby messages are
sent to the promise which should be delivered to its resolution value if it is
fulfilled with a single (as opposed to breaking with an error, or fulfilled with
multiple values).

## [Promise and Resolver Objects](#promise-objects)

Promises work like regular objects in CapTP. Promises come as a pair:

- The promise object itself which represents a value.
- The resolver object which is used to provide the promise with its resolved
  value, or break it in the case of an error.

 The promise object may eventually resolve to either a concrete value, object
 reference, another promise (in the case of promise pipelining), or may break.

>  When a promise breaks its resolved with an error, breakages can be caused by

Reviewer: ```suggestion
 When a promise breaks it is resolved with an error. Breakages can be caused by
```

>  either explicit instruction, by implicit error propagation, or network

Reviewer: ```suggestion
 either explicit instruction, by implicit error propagation, or by network
```

The `session-pubkey` is the public key part of the per-session key pair
generated for this connection. This is serialized in accordance with
[Cryptography](#cryptography).

The `acceptable-location` is a OCapN Locator which represents the location where
the session is accessible.

The `acceptable-location-sig` is a cryptographic signature of

> `acceptable-location`, wrapped with Syrup record with the label `my-location`.

Reviewer: ```suggestion
`acceptable-location`, wrapped with a Syrup record with the label `my-location`.
```

The signature is created using the private key from the per-session key pair.
This is serialized in accordance with [Cryptography](#cryptography).

### Receiving

The `captp-version` MUST be equal to `1.0`. If the version does not match, the
connection MUST be aborted.

The `acceptable-location-sig` MUST be valid that the `session-pubkey` provided a

> valid signature of `acceptable-location` wrapped with the Syrup record with the

Reviewer: ```suggestion
valid signature of `acceptable-location` wrapped with a Syrup record with the
```
